### ISSUE TITLE
PHP Error: illegal string offset foreach loop

### ISSUE THREAD

**Issue by danuri** [veritrans/veritrans-php#28]
Hi,

Iam using codeigniter and use it as library.

in VtDirect line: 18
```
if (array_key_exists('item_details', $params)) {
      $gross_amount = 0;
      foreach ($params['item_details'] as $item) {
        $gross_amount += $item['quantity'] * $item['price'];
      }
      $payloads['transaction_details']['gross_amount'] = $gross_amount;
    }
```
is getting error when single item.
- Illegal string offset 'quantity'
- Illegal string offset 'price'

And it work when I change code with:
```
if (array_key_exists('item_details', $params)) {
      $gross_amount = 0;
      foreach ($params['item_details'] as $item) {
        $gross_amount += $params['item_details']['quantity'] * $params['item_details']['price'];
      }
      $payloads['transaction_details']['gross_amount'] = $gross_amount;
    }
```

In basic programming PHP (Without CI) it run awesome without Error.
I don't know what mistake I do.

Thanks

**Comment by robeth** [comment]
Hi @danuri, thanks for using `veritrans-php` :smile:

Can you give more information about your `charge parameter` object?

I suspect something is wrong with  your `charge parameter` structure. This error is usually found when we try to access key that doesn't exist. 

I imagine that your `charge parameter` is something like this:

``` php
$charge_parameter = array(
  "payment_type" => "vtweb",
  ...
  "item_details" => array(
    "id" => "IT1",
    "name" => "Item 1",
    "quantity" => 1,
    "price" => 5000
  ),
  ...
);
``` 

While the correct `charge parameter` is:
``` php
$charge_parameter = array(
  "payment_type" => "vtweb",
  ...
  "item_details" => array(
    array(
      "id" => "IT1",
      "name" => "Item 1",
      "quantity" => 1,
      "price" => 5000
    ),
    array(
      "id" => "IT2",
      "name" => "Item 2",
      "quantity" => 1,
      "price" => 10000
    )
  ),
  ...
);
``` 

**TL;DR**
`item_details` is not _an item_ object. It is _an array_ of item objects.
